lib/Zonemaster/Backend/TestAgent.pm

```diff
@@ -158,6 +102,56 @@ sub run {
         Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, ( $params->{ipv6} ) ? ( 1 ) : ( 0 ) );
     }
 
+    my %methods = Zonemaster::Engine->all_methods;
+
+    # BASIC methods are always run
+    $counter_for_progress_indicator{planned} = {
+        'Basic::basic00' => 1,
+        'Basic::basic01' => 1,
+        'Basic::basic02' => 1,
+        'Basic::basic03' => 1,
+        'Basic::basic04' => 1,
+    };
+    foreach my $module ( keys %methods ) {
+        foreach my $method ( @{ $methods{$module} } ) {
+            if ( Zonemaster::Engine::Util::should_run_test( $method ) ) {
+                $counter_for_progress_indicator{planned}{ $module . '::' . $method } = 1;
+            }
+        }
+    }
+
+    my @planned_methods = sort keys %{ $counter_for_progress_indicator{planned} };
+
+    # used for progress indicator
+    my ( $previous_module, $previous_method ) = ( '', '' );
+
+    if ( $show_progress ) {
+        # Callback defined here so it closes over the setup above.
+        Zonemaster::Engine->logger->callback(
+            sub {
+                my ( $entry ) = @_;
+
+                if ( $entry->{tag} and $entry->{tag} eq 'TEST_CASE_END' ) {
```

Comment:
Then it must be ensured that TEST_CASE_END does not change.

Reply:
TEST_CASE_END was requested by https://github.com/zonemaster/zonemaster-engine/issues/665 and added with https://github.com/zonemaster/zonemaster-engine/pull/764.
As with everything from any other components, there might be some change in the future. And we should pay attention to such evolution to not break things in other components.
It would be great if the "TEST_CASE_END" message does not change to much indeed. And so far I don't see why it would. Are you expecting it to change soon? If so could you point to any related issue or PR or discussion about such future change, or explain how you would expect it to change?

Reply:
It was just a comment. So far it has only been meant for human reading.